java/code/src/com/redhat/rhn/taskomatic/task/MinionActionExecutor.java

```diff
@@ -192,4 +201,13 @@ private long countQueuedServerActions(Action action) {
                      .filter(serverAction -> ActionFactory.STATUS_QUEUED.equals(serverAction.getStatus()))
                      .count();
     }
+
+    private boolean allServerActionsFinished(Action action) {
+        return action != null &&
+            !CollectionUtils.isEmpty(action.getServerActions()) &&
+            action.getServerActions().stream().noneMatch(serverAction ->
```

Comment:
i think changing the condition here to all match completed or failed better matches the description of the function and is a bit easier to understand quickly.